scripts.js

```diff
@@ -0,0 +1,423 @@
+		this.option1 = this.isShifted ? 'shift' : 'default';
+		this.option2 = this.isCapsed ? 'caps' : 'regular';
+		this.layout = [...layout[this.language][this.option1][this.option2]];
+		this.keyboard = document.createElement('div');
+		this.keyboard.className = 'keyboard';
+		this.keys = [];
+	}
+
+	createLayout() {
+		this.layout.forEach((subArr, rowIdx) => {
+			let row = document.createElement('div');
+			row.className = 'row';
+			subArr.forEach((el, elIdx) => {
+				let key = document.createElement('div');
+				key.innerHTML = this.layout[rowIdx][elIdx];
+				key.className = 'key';
+				row.append(key);
+				this.keys.push(key);
+			});
+			this.keyboard.append(row);
+		})
+		document.body.append(this.keyboard);
+		return;
+	}
+
+	shift() {
+		this.isShifted = !this.isShifted;
+		this.changeLayout(this.language, this.isShifted, this.isCapsed);
+		return;
+	}
+
+	changeLanguage() {
+		this.language = this.language === 'rus' ? 'eng' : 'rus';
+		this.changeLayout(this.language, this.isShifted, this.isCapsed);
+		localStorage.setItem('language', this.language);
+	}
+
+	changeLayout(language, isShifted, isCapsed) {
+		this.isShifted = isShifted;
+		this.option1 = this.isShifted ? 'shift' : 'default';
+		this.isCapsed = isCapsed;
+		this.option2 = this.isCapsed ? 'caps' : 'regular';
+		this.language = language;
+		this.layout = [...layout[this.language][this.option1][this.option2]];
+		this.keys.forEach((key, i) => {
+			if (i < keysNum[0])	key.innerHTML = this.layout[0][i];
+			else if (i < keysNum[0] + keysNum[1]) {
+				key.innerHTML = this.layout[1][i - keysNum[0]];
+			} else if (i < keysNum[0] + keysNum[1] + keysNum[2]) {
+				key.innerHTML = this.layout[2][i - keysNum[0] - keysNum[1]];
+			}	else if (i < keysNum[0] + keysNum[1] + keysNum[2] + keysNum[3]) {
+				key.innerHTML = this.layout[3][i - keysNum[0] - keysNum[1] - keysNum[2]];
+			} else {
+				key.innerHTML = this.layout[4][i - keysNum[0] - keysNum[1] - keysNum[2] - keysNum[3]];
+			}
+		});
+		return;
+	}
+
+	shiftHandler() {
+		if (this.isShifted) {
+			this.keys[keysCode.indexOf('ShiftLeft')].classList.remove('active');
+			this.keys[keysCode.indexOf('ShiftRight')].classList.remove('active');
+			this.shift();
+		}
+	}
+
+	ctrlHandler() {
+		if (this.isShifted) {
+			this.shiftHandler();
+			this.changeLanguage();
+		}
+	}
+
+	capsHandler() {
+		this.isCapsed = !this.isCapsed;
+		this.keys[keysCode.indexOf('CapsLock')].classList.toggle('active');
+		this.changeLayout(this.language, this.isShifted, this.isCapsed);
+		return;
+	}
+}
+
+let keyboard = new Keyboard();
+
+window.onload = function() {
+	let textarea = document.createElement('TEXTAREA');
+	textarea.value = '';
+	textarea.classList = 'textarea';
+	document.body.append(textarea);
+	keyboard.createLayout();
+}
+
+function addAnimation(element) {
+	element.style.animation = '';
+	void element.offsetWidth;
+	element.style.animation = 'active 0.5s linear';
+}
+
+document.body.addEventListener('click', function(event) {
```

Comment:
The handler is too big and difficult to support. You can split it into a set of neat finctions